### ISSUE TITLE
No way to set text label on navigation button for ToolbarAndroid

### ISSUE THREAD

**Issue by ndarilek** [facebook/react-native#5312]
As an Android user who relies on a screen reader, I'd like to set a text label on the navigation button for <DrawerLayoutAndroid/>. As of now it is just an unlabeled button and I see no way to change that.

I don't know how drawer layouts typically appear, if the button is textual or if it just has a contentDescription. So I don't know what specifically to recommend, but I'd at least like the ability to add a text/accessibility label to the navigation icon while still making it look good.

Thanks.

**Comment by afnanenayet** [comment]
I'm sorry, which navigation button? I am a little unclear because you said "on the navigation button for . "

**Comment by ndarilek** [comment]
Actually, sorry, it isn't DrawerLayoutAndroid as I claimed in the title, but ToolbarAndroid.

As a TalkBack user, the navigation button in the top left doesn't have a label and is just presented as a button. It'd be nice if the ToolbarAndroid component let me assign text and an accessibility label to this button in addition to the icon image.

Thanks.

**Comment by ndarilek** [comment]
Any possibility that we'll see a fix for this sometime soon? I'm a blind developer preparing to release a React Native app to blind users, and I'm going to get crucified if it ships with an unlabeled button. :)

**Comment by ide** [comment]
@ndarilek to set expectations - if you need a feature urgently you should make it happen to meet your schedule, whether that's by writing the code yourself or hiring someone with the expertise and then building your app with your fork.

**Comment by ndarilek** [comment]
Sure, but this is also an accessibility issue that makes React Native 
less accessible across every app built with it using this component. 
Does Facebook really lack the time to improve the framework's 
accessibility? Also, surely this already works internally and isn't 
open-sourced yet, unless Facebook is really releasing apps with this 
accessibility deficiency?

For context, I'm almost always told to implement accessibility features 
myself. Maybe that's OK for some small scrappy startup, but I think 
*Facebook* should hire someone to fix this one.

**Comment by ide** [comment]
I don't work at Facebook and am not speaking on their behalf, but having some understanding of how the company operates and what all the people their working on RN are trying to accomplish, they are improving development of their own mobile apps as a means to grow the business. Most of the time this is really good because it means that RN's development is guided by real, in-house use cases in a tight feedback loop with empathy for product developers, rather than framework developers gathering feedback from external customers, but sometimes your needs aren't lined up with FB's needs.

Specifically in this case, I believe what's going on is that FB doesn't use an RN-controlled ToolbarAndroid in their flagship app, where accessibility is indeed important. Since they don't have a need for it today but do have other needs, it's a matter of priorities and there are a lot of them because the surface area of mobile APIs is so large.

The point of my previous comment is that you'll probably have a better experience going into RN with the expectation that sometimes you'll have to build features that are missing. And it's much less stressful and empowering when you can do this yourself on a schedule that you control, instead of needing someone else to do it for you. That's what we do, and sometimes when we feel our patches are robust and general enough that other people would want them, we send PRs upstream and contribute back.

**Comment by ndarilek** [comment]
This is a fair response, and I apologize if I seem snappish. FWIW, I'm 
very aware of this dynamic. For context, I discover accessibility issues 
in multiple pieces of software per week. Some are minor, some render the 
software entirely unusable. Far from being empowering, me having to fix 
all of these issues would basically mean I did nothing else. I can 
generally bang out fixes easily for HTML/JS as it's what I know best, 
but other languages/platforms/stacks are a different story.

Also, there's the whole "give a man a fish" bit. If I fix others' access 
issues, the assumption has traditionally been that I will do so going 
forward. If I point out the issues, then I can hope to create a culture 
of not having said issues introduced in future code.

In this instance, I've never used a Toolbar Android widget. I see how to 
set the `contentDescription` property on the Android widget 
programatically, but that's only a tiny fraction of the solution. 
There's also introducing the attribute on the JS side, mapping it 
idiomatically through the Java code, documenting it, possibly writing 
tests, learning about the specific test frameworks used here along with 
how the view code is mapped through the native layer...and these are all 
things I need to learn how to do for React Native, only to have to do it 
all over again Thursday for FooBarWeb, and next Wednesday for 
GitsterHipsterHosting, and the week after for SlugchatIM...see where I'm 
going? :) In the end it just seems far more efficient for someone who 
already knows this stack, this testing framework, etc. to just do it, 
then hopefully when a new component with a button is introduced in a few 
months, they'll think "ah, let's give this one a label right out the 
gate" that time. Learning is certainly a good thing but I can only 
handle so much of it.

Anyhow, it's far from empowering chasing after other people's 
accessibility issues. I was just hoping that someone who knew the stack 
could knock this out in 10-15 minutes instead of my 4-5 hours or more. :)

**Comment by lacker** [comment]
Hey, it's been a while and the right solution here still seems unclear to me. If this is just a case of adding a couple extra props that pass through to the native layer, then I think we could make progress with a list of precisely what those are and what they would do. Also you might try using some other navigation libraries - that's probably what I'd recommend in practice. Or if someone has a PR to improve this, that would be the best! Anyway, I am going to close this issue, but folks should feel free to reopen a new one if there is a specific missing prop or pull request ideally.
